src/integrationTest/java/uk/gov/hmcts/reform/professionalapi/UpdateOrganisationTest.java

```diff
@@ -0,0 +1,228 @@
+
+    @Before
+    public void setUp() {
+        professionalReferenceDataClient = new ProfessionalReferenceDataClient(port);
+        dxAddressRepository.deleteAll();
+        contactInformationRepository.deleteAll();
+        professionalUserRepository.deleteAll();
+        paymentAccountRepository.deleteAll();
+        organisationRepository.deleteAll();
+    }
+
+    @Test
+    public void updates_non_existing_organisation_returns_status_404() {
+        updateAndValidateOrganisation(UUID.randomUUID().toString(),OrganisationStatus.ACTIVE,404);
+    }
+
+    @Test
+    public void updates_organisation_with_organisation_identifier_null_returns_status_400() {
+        updateAndValidateOrganisation(null,OrganisationStatus.ACTIVE,400);
+    }
+
+    @Test
+    public void updates_organisation_with_invalid_organisation_identifier_returns_status_400() {
+        updateAndValidateOrganisation("1234ab12",OrganisationStatus.ACTIVE,400);
+    }
+
+    @Test
+    public void can_update_organisation_status_from_pending_to_active_should_returns_status_200() {
+        updateAndValidateOrganisation(createOrganisationRequest(),OrganisationStatus.ACTIVE,200);
+    }
+
+    @Test
+    public void can_update_organisation_status_from_active_to_blocked_should_returns_status_200() {
+        String organisationIdentifier = createOrganisationRequest();
+        updateAndValidateOrganisation(organisationIdentifier,OrganisationStatus.ACTIVE,200);
+        updateAndValidateOrganisation(organisationIdentifier,OrganisationStatus.BLOCKED,200);
+    }
+
+    @Test
+    public void can_update_organisation_status_from_active_to_deleted_should_returns_status_200() {
+        String organisationIdentifier = createOrganisationRequest();
+        updateAndValidateOrganisation(organisationIdentifier,OrganisationStatus.ACTIVE,200);
+        updateAndValidateOrganisation(organisationIdentifier,OrganisationStatus.DELETED,200);
+    }
+
+    @Test
+    public void can_update_organisation_status_from_pending_to_pending_should_returns_status_200() {
+        String organisationIdentifier = createOrganisationRequest();
+        updateAndValidateOrganisation(organisationIdentifier,OrganisationStatus.PENDING,200);
+    }
+
+    @Test
+    public void can_update_organisation_status_from_active_to_active_should_returns_status_200() {
+        String organisationIdentifier = createOrganisationRequest();
+        updateAndValidateOrganisation(organisationIdentifier,OrganisationStatus.ACTIVE,200);
+        updateAndValidateOrganisation(organisationIdentifier,OrganisationStatus.ACTIVE,200);
+    }
+
+    @Test
+    public void can_update_organisation_status_from_blocked_to_blocked_should_returns_status_200() {
+        String organisationIdentifier = createOrganisationRequest();
+        updateAndValidateOrganisation(organisationIdentifier,OrganisationStatus.BLOCKED,200);
+        updateAndValidateOrganisation(organisationIdentifier,OrganisationStatus.BLOCKED,200);
+    }
+

```

Comment:
A Test for Deleted to Deleted is missing

Reply:
In Jira its not mentioned but still added.